kernel/src/scan/file_stream.rs

```diff
@@ -76,7 +77,7 @@ impl DataVisitor for AddRemoveVisitor {
 }
 
 lazy_static! {
-    static ref SCAN_ROW_SCHEMA: DataType = DataType::Struct(Box::new(StructType::new(vec!(
+    pub(crate) static ref SCAN_ROW_SCHEMA: Arc<StructType> = Arc::new(StructType::new(vec!(
```

Comment:
yeah, there's lots of little cleanups like that we need to do. for instance in this PR i had the DV code take an `&Url` instead of a `Url` in `read` which let us remove a bunch of clones.